datafusion/common/src/scalar.rs

```diff
@@ -3496,6 +3496,22 @@ impl ScalarValue {
                 .map(|sv| sv.size() - std::mem::size_of_val(sv))
                 .sum::<usize>()
     }
+
+    /// This function returns the next/previous value depending on the `DIR` value.
+    /// If `true`, it returns the next value; otherwise it returns the previous value.
+    pub fn next_value<const DIR: bool>(self) -> ScalarValue {
```

Comment:
Should we add this to the public api of `ScalarValue` Seems quite specific for the usage, so it might be better to add it to the interval module

Reply:
I think I gave the opposite feedback in my initial review . I am sorry :(